zerver/views/invite.py

```diff
@@ -18,19 +18,41 @@
 
 @has_request_variables
 def invite_users_backend(request: HttpRequest, user_profile: UserProfile,
-                         invitee_emails_raw: str=REQ("invitee_emails"),
+                         invitee_emails_raw: Optional[str]=REQ("invitee_emails", default=None),
                          invite_as_admin: Optional[bool]=REQ(validator=check_bool, default=False),
-                         ) -> HttpResponse:
+                         stream_names: List[Text]=REQ(validator=check_list(check_string), default=[]),
+                         is_multiuse: Optional[bool]=REQ(validator=check_bool, default=False)) \
+        -> HttpResponse:
 
     if user_profile.realm.invite_by_admins_only and not user_profile.is_realm_admin:
         return json_error(_("Must be a realm administrator"))
+
+    streams = []  # type: List[Stream]
+    if is_multiuse:
+        if invitee_emails_raw:
+            return json_error(_("Invalid parameter 'invitee_emails'."))
+
+        if invite_as_admin:
+            return json_error(_("Users cannot be invited as admin using mutiuse link."))
+
+        for stream_name in stream_names:
+            try:
+                (stream, recipient, sub) = access_stream_by_name(user_profile, stream_name)
+            except JsonableError:
+                return json_error(_("Stream '%s' does not exist. No invites were sent.") % (stream_name,))
+            streams.append(stream)
+
+        invite_link = do_create_multiuse_invite_link(user_profile, streams)
+        return json_success({'invite_link': invite_link})
```

Comment:
You know, looking at this, I think I was probably wrong to suggest that we should have a single endpoint for these 2 use cases; what do you think?  If you agree, we should probably go back to the other way...